draft-nygren-dnsop-svcb-httpssvc.md

```diff
@@ -627,6 +627,28 @@ in cache before performing any followup queries.
 With these optimizations in place, and conforming DNS servers,
 using SVCB does not add network latency to connection setup.
 
+Moreover, if an A or AAAA response arrives before an SVCB response, the
+client SHOULD wait up to CD milliseconds before connecting as if the
+SVCB query returned NODATA, but MUST NOT transmit any information that
+could be altered by an SVCB response until the SVCB response arrives.
+For example, a TLS connection may be altered by the "esnikeys" value
+of an SVCB response. CD (Connection Delay) is a configurable parameter.
+The recommended value is 50 milliseconds, as per the guidance in {{!HappyEyeballsV2=RFC8305}}.
+
+If an SVCB record is consistent with an active or in-progress connection,
+the client may continue using that connection using any information provided
+by the SVCB RR. For example, if an SVCB record with a "esnikeys" value and
+"ipv4hint" or "ipv6hint" values arrives and one of those hints matches the
+address of an in-progress connection, the TLS connection should proceed using
+the provided ESNI keying material.
+
+If the SVCB record is not consistent with any active or in-progress connection,
+e.g. there is no "ipv4hint" or "ipv6hint" that matches a connection, clients must
```

Comment:
I would prefer a definition of "consistent" where the SVCB record is also consistent with a socket if the AAAA for SvcDomainName matches the socket's IP.

Reply:
Agreed -- I made an attempt at adding one. Please have a look!